packages/gradle-plugin/src/main/kotlin/io/realm/gradle/RealmAnalytics.kt

```diff
@@ -0,0 +1,276 @@
+// your end-user's devices; it will only run when you build your app from source.
+//
+// Why are we doing this? Because it helps us build a better product for you.
+// None of the data personally identifies you, your employer or your app, but it
+// *will* help us understand what Realm version you use, what host OS you use,
+// etc. Having this info will help with prioritizing our time, adding new
+// features and deprecating old features. Collecting an anonymized bundle &
+// anonymized MAC is the only way for us to count actual usage of the other
+// metrics accurately. If we don't have a way to deduplicate the info reported,
+// it will be useless, as a single developer building their app on Windows ten
+// times would report 10 times more than a single developer that only builds
+// once from Mac OS X, making the data all but useless. No one likes sharing
+// data unless it's necessary, we get it, and we've debated adding this for a
+// long long time. Since Realm is a free product without an email signup, we
+// feel this is a necessary step so we can collect relevant data to build a
+// better product for you.
+//
+// Currently the following information is reported:
+// - What version of Realm is being used
+// - What OS you are running on
+// - An anonymized MAC address and bundle ID to aggregate the other information on.
+
+private const val CONNECT_TIMEOUT = 4000L
+private const val READ_TIMEOUT = 2000L
+private const val TOKEN = "ce0fac19508f6c8f20066d345d360fd0"
+private const val EVENT_NAME = "Run"
+private const val URL_PREFIX = "https://webhooks.mongodb-realm.com/api/client/v2.0/app/realmsdkmetrics-zmhtm/service/metric_webhook/incoming_webhook/metric?data="
+
+internal class RealmAnalytics : TaskExecutionAdapter() {
+    companion object {
+        @Volatile
+        var METRIC_PROCESSED = false // prevent duplicate reports being sent from the same build run
+    }
+
+    override fun afterExecute(task: Task, state: TaskState) {
+        if (!state.skipped && task.name.startsWith("compile")) {
+            sendMetricIfNeeded(task.project)
+        }
+    }
+
+    private fun sendMetricIfNeeded(project: Project) {
+        if (!METRIC_PROCESSED) {
+            val disableAnalytics: Boolean = project.gradle.startParameter.isOffline || "true".equals(System.getenv()["REALM_DISABLE_ANALYTICS"], ignoreCase = true)
+            if (!disableAnalytics) {
+                val json = jsonPayload(project)
+                sendAnalytics(json)
+            }
+            METRIC_PROCESSED = true
+        }
+    }
+
+    private fun jsonPayload(project: Project): String {
+        val userId: String = ComputerIdentifierGenerator.get()
+        val appId: String = anonymousAppId(project)
+        val osType = System.getProperty("os.name")
+        val osVersion = System.getProperty("os.version")
+
+        return """{
+                   "event": "$EVENT_NAME",
+                   "properties": {
```

Comment:
In Java we also attempt to track Android minSdk/targetSdk, would it make sense to attempt to do this here as well?

It would probably also be nice to see if people are consuming sync/non-sync and if it is a KMP/pure android project. But since we are not supporting that just yet, we can probably just postpone it?

Reply:
Added min/target SDK. We'll add Sync when it's supported